PyISY/Nodes/__init__.py

```diff
@@ -141,6 +141,20 @@ def _upmsg(self, xmldoc):
         self.getByID(nid).status.update(nval, force=True, silent=True)
         self.parent.log.info('ISY Updated Node: ' + nid)
 
+    def _controlmsg(self, xmldoc):
+        """Passes Control events from an event stream message to nodes, for
+        sending out to subscribers."""
+        try:
+            nid = xmldoc.getElementsByTagName('node')[0].firstChild.toxml()
+            cntrl = xmldoc.getElementsByTagName('control')[0].firstChild.toxml()
+            self.getByID(nid).controlEvents.notify(cntrl)
+            self.parent.log.info('ISY Node Control Event: ' + nid + ' ' + cntrl)
+        except:
```

Comment:
I understand what you're doing here, but is there a better way to ensure the XML contains the necessary elements without using try/except? Possibly:

```python
nodes = xmldoc.getElementsByTagName('node')
ctrls = xmldoc.getElementsByTagName('control')

if len(nodes) > 0 and len(cntrls) > 0:
    nid = nodes[0].firstChild.toxml()
    cntrl = cntrls[0].firstChild.toxml()
    self.getByID(nid).controlEvents.notify(cntrl)
    self.parent.log.info('ISY Node Control Event: ' + nid + ' ' + cntrl)
```

Reply:
It most definitely shouldn't be a catchall `except`, but when looking around it seemed to be a common belief that the Pythonic way to handle checking if something exists is to just try it and catch it, which is why I went with this pattern. I can change it to however you'd like it to be though

Reply:
Meh, I'm kind of 50-50 on this one. You can keep it how it is, but a more specific except clause would be good.